modules/ROOT/pages/dataweave-type-reuse-xml-schema.adoc

```diff
@@ -0,0 +1,429 @@
+| `xs:long`| `Number`
+| `xs:short`| `Number`
+| `xs:byte`| `Number`
+| `xs:double`| `Number`
+| `xs:decimal`| `Number`
+| `xs:unsignedLong`| `Number`
+| `xs:unsignedInt`| `Number`
+| `xs:unsignedShort`| `Number`
+| `xs:unsignedByte`| `Number`
+| `xs:positiveInteger`| `Number`
+| `xs:negativeInteger`| `Number`
+| `xs:nonNegativeInteger`| `Number`
+| `xs:nonPositiveInteger`| `Number`
+| `xs:hexBinary`| `Binary`
+| `xs:base64Binary`| `Binary`
+| `xs:boolean`| `Boolean`
+| `xs:any`| `String`
+| `xs:string`| `String`
+| `xs:normalizedString`| `String`
+| `xs:datetime`| `DateTime`
+| `xs:time`| `Time`
+| `xs:date`| `LocalDateTime`
+|===
+
+.XSD:
+[source,XML]
+----
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<xs:schema attributeFormDefault="unqualified" elementFormDefault="qualified"  xmlns:xs="http://www.w3.org/2001/XMLSchema">
+    <xs:element name="Root">
+        <xs:complexType>
+            <xs:all>
+                <xs:element name="string" type="xs:string"/>
+                <xs:element name="normalizedString" type="xs:normalizedString"/>
+
+
+                <xs:element name="integer" type="xs:integer"/>
+                <xs:element name="negativeInteger" type="xs:negativeInteger"/>
+                <xs:element name="int" type="xs:int"/>
+                <xs:element name="unsignedInt" type="xs:unsignedInt"/>
+                <xs:element name="unsignedLong" type="xs:unsignedLong"/>
+                <xs:element name="unsignedShort" type="xs:unsignedShort"/>
+                <xs:element name="float" type="xs:float"/>
+                <xs:element name="double" type="xs:double"/>
+
+
+                <xs:element name="date" type="xs:date"/>
+                <xs:element name="dateTime" type="xs:dateTime"/>
+                <xs:element name="time" type="xs:time"/>
+
+
+                <xs:element name="boolean" type="xs:boolean"/>
+
+
+                <xs:element name="hexBinary" type="xs:hexBinary"/>
+                <xs:element name="base64Binary" type="xs:base64Binary"/>
+                <xs:element name="decimal" type="xs:decimal"/>
+
+
+
+
+
+
+
+
+            </xs:all>
+        </xs:complexType>
+    </xs:element>
+</xs:schema>
+----
+
+.Generated DataWeave Module:
+[source,dataweave,linenums]
+----
+%dw 2.0
+
+
+type RootElementDefinition = {| Root: { string: String, normalizedString: String, integer: Number, negativeInteger: Number, int: Number, unsignedInt: Number, unsignedLong: Number, unsignedShort: Number, float: Number, double: Number, date: LocalDateTime, dateTime: DateTime, time: Time, boolean: Boolean, hexBinary: Binary, base64Binary: Binary, decimal: Number } |}
+
+
+type Root = RootElementDefinition
+----
+
+== XSD Structure Mapping
+
+=== Root Elements Declaration
+
+Each root element declared inside your XSD file generates a type directive with the element name followed by the `ElementDeclaration` extension and a type directive named *Root*, considered a Union type, consisting of all your root elements.
```

Comment:
Making a suggestion here. What do you think?

Each root element (`<xs:element/>`) declared inside your XSD file generates the following:
* A DataWeave `type` directive with the element name followed by the `ElementDeclaration` extension.
* A `type` directive named `Root` as a `Union` type that consists of all the root elements in the XSD.
@martincousido

Reply:
Yes! I agree with that suggestion. The only change would be to put <xs:element> instead of <xs:element/> (without the "/" slash)